json_rpc/router.nim

```diff
@@ -71,17 +81,30 @@ proc route*(router: RpcRouter, node: JsonNode): Future[StringOfJson] {.async, gc
     return wrapError(INVALID_REQUEST, "'method' missing or invalid")
 
   let rpcProc = router.procs.getOrDefault(methodName)
-  if rpcProc == nil:
+  let proxyCall = router.proxyCalls.getOrDefault(methodName)
+  let params = node.getOrDefault("params")
+
+  if rpcProc == nil and (proxyCall == nil):
     return wrapError(METHOD_NOT_FOUND, "'" & methodName & "' is not a registered RPC method", id)
+  elif rpcProc == nil and (proxyCall != nil):
+    try:
+      let res = await proxyCall(methodName, if params == nil: newJArray() else: params)
```

Comment:
Regarding your question of error handling in the recovery/proxy procedure, I don't think we have a stance on this? It probably depends on the situation, but considering this is a networked (async) request and you need its return value to proceed, it probably fits better here the way you did it. (It would be different in other situations e.g. where you use `asyncSpawn`).

Reply:
that was my line of thinking, that from router pov this procedure could be anything (in our case it is network call), so router needs to handle all errors this procedure can raise.